compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -143,6 +144,12 @@ private class ExtractAPICollector(using Context) extends ThunkHolder {
     */
   private val refinedTypeCache = new mutable.HashMap[(api.Type, api.Definition), api.Structure]
 
+  /** This cache is necessary to avoid infinite loops when hashing the body of inline definitions.
+   *  Its keys represent the root inline definitions, and its values are seen inline references within
+   *  the rhs of the key. If a symbol is present in the value set, then do not hash its signature or inline body.
+   */
+  private val seenInlineCache = mutable.HashMap.empty[Symbol, mutable.HashSet[Symbol]]
```

Comment:
Instead of a Map, could the cache be a `mutable.HashSet[Symbol]` which we `clear()` at the end of apiAnnotations if inlineOrigin.isEmpty?